custom_components/nrgkick/websocket.py

```diff
@@ -187,6 +188,27 @@ async def set_charge_current_limit(self, limit: float) -> bool:
             return False
         return True
 
+    async def set_energy_current_limit(self, limit: float) -> bool:
+        """Set the energy current limit."""
+        event = asyncio.Event()
+
+        request = self.__get_base_request()
+        request.header.type = nrgcp.Nrgcp.Header.Type.UPDATE
+        request.header.service = nrgcp.Nrgcp.Header.Service.CHARGE_CONTROL
+        request.header.property = nrgcp.Nrgcp.Header.Property.SETTINGS
+
+        request.payload.CHARGECONTROL_SETTINGS_UPDATE.energyLimit.limited = NrgcpTypes.EnergyLimitMode.ENERGY_LIMIT_MODE_LIMITED
```

Comment:
It would be nice if we had a way to set it back to unlimited some way.

Reply:
Added a new switch to control ENERGY_LIMIT_MODE